### PR TITLE
Fixed code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
This pull request fixes some of the code quality issues raised by DeepSource on my fork of this repository.
I have already fixed some issues using DeepSource's Autofix.

Take a quick look at all the issues caught by DeepSource for this repository [here](https://deepsource.io/gh/withshubh/improver/issues/?category=recommended).

### Summary of fixes

- Remove unnecessary use of comprehension
- Remove unused imports
- Use literal syntax instead of function calls to create data structure
- Remove unnecessary generator
- Remove unnecessary `return` statement

You can also have a look at the [configuration file](https://github.com/withshubh/improver/blob/deepsource/.deepsource.toml) I used for DeepSource Analysis.

CLA
 - [x] If a new developer, signed up to CLA

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Thank you @withshubh! We appreciate the contribution. Our tests vs black etc are broken in this PR - could you please fix it? It might take us a bit of time to review the PR after that due to the number of lines and a bit of a rush on at the moment, and we'll need to look a bit at DeepSource.

**Comment by withshubh** [comment]
> Thank you @withshubh! We appreciate the contribution. Our tests vs black etc are broken in this PR - could you please fix it? It might take us a bit of time to review the PR after that due to the number of lines and a bit of a rush on at the moment, and we'll need to look a bit at DeepSource.

Hey @benfitzpatrick :wave: 
No worries. I'll fix the broken tests and update the PR. 😄

**Comment by withshubh** [comment]
@benfitzpatrick :wave: I pushed a commit reverting the changes that were breaking tests. Please have a look :eyes:

**Comment by withshubh** [comment]
Hey @benfitzpatrick :wave: 

Please have a look :eyes:

**Comment by withshubh** [comment]
@benfitzpatrick Please have a :eyes: at this PR! :sparkles:

**Comment by withshubh** [comment]
@MoseleyS Can you review this PR! :eyes:

**Comment by benfitzpatrick** [comment]
Sincere apologies @withshubh - I have been snowed under with other work and haven't got to this. I can start going through this, but there are unfortunately still some conflicts to resolve.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ below:
 * Eleanor Smith (Met Office, UK)
 * Tomasz Trzeciak (Met Office, UK)
 * Mark Worsfold (Met Office, UK)
+* Shubhendra Singh Chauhan (DeepSource, India)
```
Comment (benfitzpatrick) [inline CONTRIBUTING.md:64]: Very happy for you to put your name with all the others in alphabetical ordering

**Review by benfitzpatrick — COMMENTED** [review]
Thank you @withshubh. There are a few conflicts to resolve but otherwise looks good.

**Review by MoseleyS — COMMENTED** [review]
I'm happy with these changes. All our tests still pass, once the conflicts are resolved!

**Comment by MoseleyS** [comment]
@benfitzpatrick : If we want to adopt this standard, should it be activated with a Github Action?

**Comment by benfitzpatrick** [comment]
It probably should, but I need to investigate a bit

**Comment by withshubh** [comment]
> @benfitzpatrick : If we want to adopt this standard, should it be activated with a Github Action?

@MoseleyS Are you referring to DeepSource here?

let me share few things about it and make @benfitzpatrick investigation easy :wink: 

DeepSource is a code review automation tool that detects code quality issues and helps you to automatically fix some of them. You can use DeepSource to track test coverage, Detect problems in Dockerfiles, etc. in addition to detecting issues in code. 
It is a service whose checks will run in parallel to CI. Free for Open-Source forever. :sparkling_heart: 

Shall I share the steps to activate analysis to continuously analyze this repo?

**Comment by benfitzpatrick** [comment]
Thanks @withshubh! And sorry for the further delay. I think I'm happy to merge this as-is once conflicts are resolved.

**Comment by withshubh** [comment]
@benfitzpatrick I don't have access to resolve the conflicts
bitmoji

**Comment by benfitzpatrick** [comment]
Apologies, you may have to do it via the command line, not sure why GitHub won't let you - I unfortunately can't do it as the source branch is the master in your fork.

**Comment by withshubh** [comment]
@benfitzpatrick done :sparkles:

File: improver/cli/snow_fraction.py [hunk 2]
```diff
@@ -58,4 +58,6 @@ def process(*cubes: cli.inputcube, model_id_attr: str = None):
     from improver.precipitation_type.snow_fraction import SnowFraction
     from iris.cube import CubeList
 
+    from improver.snow_fraction import SnowFraction
```
Comment (benfitzpatrick) [inline improver/cli/snow_fraction.py:61]: Failure in tests linked to this - I think it might need swapping with line 58?

Comment (withshubh) [reply improver/cli/snow_fraction.py]: swapped

Comment (benfitzpatrick) [reply improver/cli/snow_fraction.py]: Ah sorry, my clumsy wording - I meant that the line `improver.snow_fraction` should be replaced with `improver.precipitation_type.snow_fraction` as that's the new location

File: improver/cli/snow_fraction.py [hunk 3]
```diff
@@ -58,4 +58,6 @@ def process(*cubes: cli.inputcube, model_id_attr: str = None):
     from improver.precipitation_type.snow_fraction import SnowFraction
```
Comment (benfitzpatrick) [inline improver/cli/snow_fraction.py:58]: Apologies @withshubh, but we now have snow fraction twice? Everything else looks good.

File: improver/cli/snow_fraction.py (hunk 2 above)
Comment (withshubh) [reply improver/cli/snow_fraction.py]: My bad I missed this.

Fixed

**Review by benfitzpatrick — APPROVED** [review]
Thank you @withshubh!

**Comment by withshubh** [comment]
@benfitzpatrick The fixes in this PR were made using DeepSource. Can you please activate the analysis and share your feedback with me?
All you need to do is signup [here](http://deepsource.io/signup) and activate analysis [here](https://deepsource.io/gh/metoppv/improver/).

**Comment by withshubh** [comment]
Hey @benfitzpatrick :wave: 

Did you activate and try out DeepSource? Would love to know your feedback :)

**Comment by benfitzpatrick** [comment]
Hi @withshubh, will look into it but we do have a couple of recent and upcoming changes to what we're testing with - will definitely bear it in mind, thanks.

**Comment by withshubh** [comment]
> Hi @withshubh, will look into it but we do have a couple of recent and upcoming changes to what we're testing with - will definitely bear it in mind, thanks.

Thanks!

**Pull request merged**